pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -83,21 +83,22 @@ def _check_consider_iterating_dictionary(self, node: nodes.Call) -> None:
             return
         if node.func.attrname != "keys":
             return
+        comp_ancestor = utils.get_node_first_ancestor_of_type(node, nodes.Compare)
         if (
             isinstance(node.parent, (nodes.For, nodes.Comprehension))
-            or isinstance(node.parent, nodes.Compare)
+            or isinstance(comp_ancestor, nodes.Compare)
             and any(
                 op
-                for op, comparator in node.parent.ops
-                if op == "in" and comparator is node
+                for op, comparator in comp_ancestor.ops
+                if op in ["in", "not in"]
```

Comment:
```suggestion
                if op in {"in", "not in"}
```

Reply:
isn't tuple a better idea here?

```suggestion
                if op in ("in", "not in")
```

Reply:
When checking for membership sets are faster, we have a check for that but it seems it's not activated right now.